sync/syncer.go

```diff
@@ -166,12 +167,28 @@ func (s *Syncer) ListenToGossip() bool {
 	return s.getGossipBufferingStatus() != Pending
 }
 
-func (s *Syncer) setGossipBufferingStatus(b Status) {
+func (s *Syncer) setGossipBufferingStatus(status Status) {
 	s.gossipLock.Lock()
-	s.gossipSynced = b
+	s.notifySubscribers(s.gossipSynced, status)
+	s.gossipSynced = status
 	s.gossipLock.Unlock()
 }
 
+func (s *Syncer) notifySubscribers(prevStatus, status Status) {
+	if (status == Done) == (prevStatus == Done) {
+		return
+	}
+	if status == Done {
+		close(s.awaitCh)
+	} else {
+		s.awaitCh = make(chan struct{})
```

Comment:
why create a channel here? this has nothing to do with notifying subscribers

Reply:
Currently subscribers just get whatever is in `s.awaitCh` which makes a simpler mechanism (I also create a channel in the constructor).

The alternative is to put a `nil` there and when someone subscribes do a nil-check and create the channel then, if needed, or return a closed channel. This would be more cumbersome.

Another advantage of the current implementation is that I don't need to lock the syncer for `Await()` calls, since it just returns whatever is in `s.awaitCh` (`notifySubscribers` is only called under lock anyway).